programs/bpf_loader/src/serialization.rs

```diff
@@ -994,58 +995,94 @@ mod tests {
     }
 
     // the old bpf_loader in-program deserializer bpf_loader::id()
-    #[allow(clippy::type_complexity)]
+    #[deny(unsafe_op_in_unsafe_fn)]
```

Comment:
personally, i want this to be enabled at tree-wide. but this is different story.